src/diffusers/models/unets/unet_2d_blocks.py

```diff
@@ -772,14 +778,17 @@ def __init__(
         if isinstance(transformer_layers_per_block, int):
             transformer_layers_per_block = [transformer_layers_per_block] * num_layers
 
+        resnet_groups_out = resnet_groups_out or resnet_groups
```

Comment:
```suggestion
```

Reply:
Because the in & out channels can be different, the `resnet_groups` and  `resnet_groups_out` can also be different. So we need this lines.